src/openforms/config/models.py

```diff
@@ -649,6 +655,33 @@ def as_dict(self):
         return {k: list(v) for k, v in ret.items()}
 
 
+class CSPSettingManager(models.Manager.from_queryset(CSPSettingQuerySet)):
+    def set_for(self, obj: models.Model, settings: list[tuple[str, str]]) -> None:
```

Comment:
I think a `@transaction.atomic` decorator would be good to guarantee that we don't end up without csp directives if the insert crashes for some reason.